### ISSUE TITLE
Categorical metrics filter widget

### ISSUE THREAD

**Issue by mindrones** [nestauk/asf_hp_market_tracker#171]
In the filters sidebar, for each categorical metric we see a widget showing the metric name and the list of available categories: these could be a list of checkboxes with their label or pills that can be (dis)activated by clicking. These should be fetched as static data using the `terms` endpoint.

When we (de)select one of the categories of a metric, the widget shows 2 icons in the top right corner to `apply` or `dismiss` the new selection:
- clicking on the `dismiss` icon brings back the categories of that metric to as they were before we started editing
- clicking on the `apply` icon updates the `_filters` store (we'll implement actual querying in another PR).

**Before selecting / After dismissing**

<img width="306" alt="Screenshot 2023-06-26 at 18 15 46" src="https://github.com/nestauk/asf_hp_market_tracker/assets/1309648/2e81006d-c765-4668-b557-dd2505b4ab9c">

**During selection**

<img width="305" alt="Screenshot 2023-06-26 at 18 15 51" src="https://github.com/nestauk/asf_hp_market_tracker/assets/1309648/b968e2de-388b-4445-b382-df2808ee3291">

**After applying**

<img width="305" alt="Screenshot 2023-06-26 at 18 15 57" src="https://github.com/nestauk/asf_hp_market_tracker/assets/1309648/af23733e-ab2e-4486-8a4d-399451dcc393">
